### DISCUSSION TITLE
Lucene terms example

### DISCUSSION THREAD

**Discussion by scottiemc7** [OrchardCMS/OrchardCore#11943] [Q&A] [answered]
Hi.

I am trying to pass a terms filter into a query, but not having any luck. Here's the query:

```
{
  "query": {
    "bool" : {
        "must": {
            "match_all": {}
          },
        "filter": [
            { "terms" : { "Listing.Type" : ["{{eventType}}"] } }
          ]
    }
  },
  "sort": [
    { "Content.ContentItem.ModifiedUtc": { "order": "desc" } },
    { "Content.ContentItem.CreatedUtc": { "order": "desc" } }
  ]
}
```

And the input is just:
`{ "eventType": "Event, Club" }`

as suggested [here](https://github.com/OrchardCMS/OrchardCore/issues/3550#issuecomment-583750013), but the query returns nothing. When I embed **"Event","Club"** inline the query works as expected. How would one use a terms filter with multiple terms?

Thank you

**Comment by Skrypt** [comment]
Make sure the Listing.Type is set to "stored".

**Reply by scottiemc7** [reply]
@Skrypt thank you. Are you talking about the "Include this element in the index" option? If so, it's already included. The query works fine if I change the line

`{ "terms" : { "Listing.Type" : ["{{eventType}}"] } }`
to
`{ "terms" : { "Listing.Type" : ["Event", "Club"] } }`

which makes me wonder if this is related to liquid somehow? Perhaps I'm passing in the arguments wrong.

<img width="749" alt="Screen Shot 2022-09-11 at 3 18 20 PM" src="https://user-images.githubusercontent.com/1521455/189551247-18094735-e920-40eb-82ee-2aecf28ee44d.png">

**Reply by Skrypt** [reply]
{{event_type}}

Liquid never use camel case.

**Reply by scottiemc7** [reply]
OK thanks, that's good to know. If I change the parameter name to e like `{ "terms" : { "Listing.Type" : {{e}} } }` ,
and use input `{ "e": ["Club", "Event"] }` I get:

`invalid query : Unexpected character encountered while parsing value: C. Path 'query.bool.filter[0].terms['Listing.Type']', line 8, position 43.`

I think this should be the correct way to write the query. The other example I gave would (I think) actually end up looking like `{ "terms" : { "Listing.Type" : ["Club,Event"] } }` when what I really want is `{ "terms" : { "Listing.Type" : ["Club", "Event"] } }`

**Reply by Skrypt** [reply]
It should work. If not then report an issue with repro steps so that we can take a look at it.

**Reply by scottiemc7** [reply]
Ah OK. Thank you for the sanity check. I will report this as a possible issue.

**Reply by scottiemc7** [reply]
I just [updated the ticket](https://github.com/OrchardCMS/OrchardCore/issues/12439#issuecomment-1270989052). It was closed, but I feel like the issue is not yet solved. I discovered some more details. If you have time, would you mind commenting on it? Thank you.

**Comment by scottiemc7** [comment, marked as answer]
{{ e | json | raw }} ended up being the right way to filter the parameter. Thanks for your help.
